File: jar/src/main/java/jline/api/pfqn/mva/Pfqn_bjbk.java
/**
 * @file Iterative BJB(k) balanced job bounds (Zahorjan et al.) for single-class closed networks
 *
 * BJB(k) at iteration count k; BJB(1) recovers the noniterative Balanced Job Bound. Realized
 * through the validated pfqn_pbh recursion. Ported at parity from MATLAB pfqn_bjbk.m.
 *
 * @since LINE 3.0
 */
package jline.api.pfqn.mva;

import jline.util.matrix.Matrix;

public final class Pfqn_bjbk {
    private Pfqn_bjbk() {}

    /**
     * Iterative balanced job bounds BJB(k).
     *
     * @param L service demand vector (M x 1)
     * @param N population
     * @param Z think time
     * @param k iteration count (>= 1)
     * @return {Xlo, Xhi} lower/upper throughput bounds
     */
    public static double[] pfqn_bjbk(Matrix L, double N, double Z, int k) {
        return Pfqn_pbh.pfqn_pbh(L, N, Z, k);
    }
}

File: matlab/src/api/pfqn/pfqn_bjbk.m
%{
%{
 % @file pfqn_bjbk.m
 % @brief Iterative BJB(k) balanced job bounds (Zahorjan et al.) for
 %        single-class closed networks with delay.
%}
%}

function [Xlo,Xhi] = pfqn_bjbk(L,N,Z,k)
%{
%{
 % @brief BJB(k): the iterative Balanced Job Bound at iteration count k
 %        (Casale-Muntz-Serazzi 2008, Tables 5,7). BJB(1) recovers the
 %        noniterative Balanced Job Bound; each additional iteration performs
 %        one exact MVA step from the balanced seed at N-k, and the bracket
 %        tightens to exact as k->N. Realized through the validated PBH
 %        recursion (pfqn_pbh), whose level-1 optimistic bound equals the
 %        noniterative BJB optimistic bound (Eager-Sevcik 1983, eq. 10).
 % @fn pfqn_bjbk(L, N, Z, k)
 % @param L Service demand vector (M x 1).
 % @param N Population (scalar).
 % @param Z Think time (scalar, default 0).
 % @param k Iteration count >= 1 (default 1).
 % @return Xlo Lower throughput bound.
 % @return Xhi Upper throughput bound.
%}
%}
if nargin < 3 || isempty(Z), Z = 0; end
if nargin < 4 || isempty(k), k = 1; end
[Xlo,Xhi] = pfqn_pbh(L,N,Z,k);
end
